base/loading.jl

```diff
@@ -611,7 +611,8 @@ end
 # or an Exception that describes why it couldn't be loaded
 # and it reconnects the Base.Docs.META
 function _include_from_serialized(path::String, depmods::Vector{Any})
-    restored = ccall(:jl_restore_incremental, Any, (Cstring, Any), path, depmods)
+    sv = ccall(:jl_restore_incremental, Any, (Cstring, Any), path, depmods)
+    restored = sv[1]
```

Comment:
@Keno, this looks incorrect. The return value from `jl_restore_incremental` is not indexable?

Reply:
This same commit changed the regular return value to be an exception, but you are correct that something got messed up with the other return path in this function. Wanna fix it and ideally at a test case ;)?